packages/manager/apps/web/client/app/email-domain/order/email-domain-order.controller.js

```diff
@@ -1,176 +1,105 @@
-          order: 'mxPlan.alerts.order',
-          durations: 'mxPlan.alerts.order.duration',
-        };
-
-        this.initialLoad()
-          .then(() => {
-            if (this.selectedOrder.config.domain) {
-              this.fetchOffers();
-            }
-          });
-      }
-
-      initialLoad() {
-        this.loading.init = true;
-        this.User.getUrlOfEndsWithSubsidiary('MX_PLAN').then((page) => {
-          this.urls.mxPlanProductDescription = page;
-        });
-
-        return this.Domain.getDomains()
-          .then((domains) => { this.list.domains = domains; })
-          .catch(() => this.Alerter.error(this.$translate.instant('mxPlan_order_step1_error'), this.$scope.alerts.order))
-          .finally(() => { this.loading.init = false; });
-      }
-
-      fetchOffers() {
-        this.loading.offers = true;
-        this.selectedOrder.offer = null;
-        this.list.offers = [];
-
-        return this.MXPlan
-          .getOrderModels(this.selectedOrder.config.domain)
-          .then((response) => { this.list.offers = response; })
-          .catch((err) => {
-            this.list.offers = filter(err, (offer) => offer.offer);
-            const errors = filter(err, (offer) => offer.message);
-            if (this.list.offers.length <= 0) {
-              this.Alerter.error(this.$translate.instant('mxPlan_order_step1_error', { t0: errors[0].message }), this.$scope.alerts.order);
-            }
-          })
-          .finally(() => { this.loading.offers = false; });
-      }
-
-      selectOffer() {
-        this.selectedOrder.contractsValidated = false;
-        if (this.selectedOrder.config.domain
-          && this.selectedOrder.config.domain.length
-          && this.selectedOrder.config.offer
-          && this.selectedOrder.config.offer.length) {
-          this.order = null;
-          this.selectedOrder.duration = null;
-          this.getDurations();
+import find from 'lodash/find';
+import get from 'lodash/get';
+import includes from 'lodash/includes';
+
+import { ORDER_TRACKING_URLS } from './order.constants';
+
+export default class MXPlanOrderCtrl {
+  /* @ngInject */
+  constructor($q, $timeout, $translate, $window, Alerter, coreConfig) {
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.Alerter = Alerter;
+    this.coreConfig = coreConfig;
+  }
+
+  $onInit() {
+    this.currentIndex = 0;
+
+    this.model = {
+      selectedDomain: find(this.domains, this.domain),
+      product: this.products[0],
+      autoPayWithPreferredPaymentMethod: null,
+      contractsAccepted: false,
+    };
+
+    this.loading = {
+      domain: false,
+      checkout: false,
+    };
+  }
+
+  getProductTaxAmount({ planCode }) {
+    const { plans } = this.catalog;
+    const pricings = get(find(plans, { planCode }), 'pricings');
+    const price = find(pricings, ({ capacities }) => includes(capacities, 'installation'));
```

Comment:
We are making a lot of assumptions on UX side, like : the product price will be the price of the installation capability.

I know you're not responsible for that but i think it's a huge design flaw from the order api, we should receive on UX side a clear total price.

I even think it could backfire on us in the future (why the hell does UX compute the price of the product from the installation price only? and people will forgot that we had no choices from the begining...)